src/app/globals.css

```
@@ -120,3 +120,31 @@
     @apply text-foreground font-sans;
   }
 }
+
+.fluid-button {
+  position: relative;
+  overflow: hidden;
+  transition:
+    background-color 0.8s ease,
+    color 0.8s ease;
+}
+.fluid-button::before {
+  content: "";
+  position: absolute;
+  bottom: 0;
+  left: 50%;
+  width: 0;
+  height: 0;
+  background: rgba(0, 0, 0, 0.1);
+  border-radius: 50%;
+  transform: translate(-50%, 100%);
+  transition:
+    width 0.8s ease,
+    height 0.8s ease,
+    transform 0.8s ease;
+}
+.fluid-button:hover::before {
+  transform: translate(-50%, -50%);
+  width: 300%;
+  height: 300%;
+}
```

to jest SCSS, więc możesz to zanestować
```suggestion
.fluid-button {
	position: relative;
	overflow: hidden;
	transition:
	  background-color 0.8s ease,
	  color 0.8s ease;

	&::before {
	  content: "";
	  position: absolute;
	  bottom: 0;
	  left: 50%;
	  width: 0;
	  height: 0;
	  background: rgba(0, 0, 0, 0.1);
	  border-radius: 50%;
	  transform: translate(-50%, 100%);
	  transition:
	    width 0.8s ease,
	    height 0.8s ease,
	    transform 0.8s ease;
	}

	&:hover::before {
	  transform: translate(-50%, -50%);
	  width: 300%;
	  height: 300%;
	}
}
```